### PR TITLE
Make access-control.config-path configurable

### PR DESCRIPTION
**Pull request by maciejgrzybek**
This is to avoid hard coded path for configuration and allow some flexibility for the user deploying Presto.

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/security/AccessControlManager.java [hunk 1]
```diff
@@ -121,12 +122,12 @@ public void removeCatalogAccessControl(ConnectorId connectorId)
     public void loadSystemAccessControl()
             throws Exception
     {
-        if (ACCESS_CONTROL_CONFIGURATION.exists()) {
-            Map<String, String> properties = new HashMap<>(loadProperties(ACCESS_CONTROL_CONFIGURATION));
+        if (configurationFile.exists()) {
```
Comment (arhimondr) [inline presto-main/src/main/java/com/facebook/presto/security/AccessControlManager.java]: I think we should throw an exception when configuration file is explicitly specified, but it doesn't exists.

**Comment by arhimondr** [comment]
lgtm % comment

File: presto-docs/src/main/sphinx/develop/system-access-control.rst [hunk 2]
```diff
@@ -34,9 +34,8 @@ Configuration
 
 After a plugin that implements ``SystemAccessControl`` and
 ``SystemAccessControlFactory`` has been installed on the coordinator, it is
-configured using an ``etc/access-control.properties`` file. All of the properties
-other than ``access-control.name`` are specific to the ``SystemAccessControl``
-implementation.
+configured using a file specified in ``access-control.config-path`` property of ``config.properties`` (by default it points to ``etc/access-control.properties`` file).
```
Comment (cawallin) [inline presto-docs/src/main/sphinx/develop/system-access-control.rst]: "specified in the ``access-control.config-path``"
"(by default, ``etc/access-control.properties``)"

File: presto-main/src/main/java/com/facebook/presto/security/AccessControlConfig.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    private String accessControlConfigPath = "etc/access-control.properties";
+
+    @NotNull
+    public String getAccessControlConfigPath()
+    {
+        return accessControlConfigPath;
+    }
+
+    @Config("access-control.config-path")
+    @ConfigDescription("Path with filename to access control properties")
```
Comment (cawallin) [inline presto-main/src/main/java/com/facebook/presto/security/AccessControlConfig.java]: suggest: "Path for the access control properties"

File: presto-main/src/main/java/com/facebook/presto/security/AccessControlConfig.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ * limitations under the License.
+ */
+package com.facebook.presto.security;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+
+import javax.validation.constraints.NotNull;
+
+public class AccessControlConfig
```
Comment (cawallin) [inline presto-main/src/main/java/com/facebook/presto/security/AccessControlConfig.java]: Add a test

**Comment by dain** [comment]
I'm not sure we want to do this.  Following this pattern we should have config options for the location of all configuration files.  Instead people should use symlinks (or hard links).

**Comment by electrum** [comment]
This seems overly complex. Why not just use a symlink?

**Comment by maciejgrzybek** [comment]
Then, why do we have `plugin.dir` or `node.data-dir` configurable in `node.properties`? The same could be achieved by symlinks.

**Comment by maciejgrzybek** [comment]
Any thoughts on https://github.com/prestodb/presto/pull/6772#issuecomment-265233836, @electrum and/or @dain?

**Comment by dain** [comment]
I'd like to hear @martint's thoughts on this

**Comment by ebd2** [comment]
Closed in favor of #7077

**Pull request closed without merging**